From ed6d04ec78537e46deac42f2b3322f3532cd698b Mon Sep 17 00:00:00 2001
From: Masato TOYOSHIMA <phoepsilonix@phoepsilonix.love>
Date: Sat, 26 Oct 2024 11:14:13 +0900
Subject: [PATCH] Fcitx bazel build (#68)

  * add build_fcitx_bazel

  * This is a fix to enable fcitx-mozc to be built with bazel.
---
 scripts/build_fcitx_bazel  | 5 +++++
 src/unix/fcitx/BUILD.bazel | 4 ----
 2 files changed, 5 insertions(+), 4 deletions(-)
 create mode 100755 scripts/build_fcitx_bazel

diff --git a/scripts/build_fcitx_bazel b/scripts/build_fcitx_bazel
new file mode 100755
index 000000000..c68e9e61a
--- /dev/null
+++ b/scripts/build_fcitx_bazel
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+_BUILD_TARGETS="${_BUILD_TARGETS:-unix/fcitx:fcitx-mozc.so server:mozc_server gui/tool:mozc_tool}"
+
+bazel build -c opt --copt=-fPIC  --config oss_linux "$@" $_BUILD_TARGETS
diff --git a/src/unix/fcitx/BUILD.bazel b/src/unix/fcitx/BUILD.bazel
index 76beb768d..1b2d17f44 100644
--- a/src/unix/fcitx/BUILD.bazel
+++ b/src/unix/fcitx/BUILD.bazel
@@ -19,7 +19,6 @@ mozc_cc_library(
         "//client",
         "//protocol:commands_cc_proto",
         "@com_google_absl//absl/log:check",
-        "@fcitx",
     ],
 )
 
@@ -65,7 +64,6 @@ mozc_cc_library(
         "//protocol:commands_cc_proto",
         "//protocol:config_cc_proto",
         "@com_google_absl//absl/log:check",
-        "@fcitx",
     ],
 )
 
@@ -82,7 +80,6 @@ mozc_cc_library(
         "//base:util",
         "//base:vlog",
         "@com_google_absl//absl/log:check",
-        "@fcitx",
     ],
 )
 
@@ -97,6 +94,5 @@ mozc_cc_binary(
     deps = [
         ":fcitx_mozc",
         "//base:init_mozc",
-        "@fcitx",
     ],
 )
